formatcode.sh

```diff
@@ -14,27 +19,42 @@ set -o nounset
 # Get CPU count
 OS=$(uname)
 NPROC=1
-if [[ $OS = "Linux" || $OS = "Darwin" ]] ; then
-    NPROC=$(getconf _NPROCESSORS_ONLN)
+if [[ ${OS} = "Linux" ]] ; then
+    NPROC=$(nproc)
+elif [[ ${OS} = "Darwin" ]] ; then
+    NPROC=$(sysctl -n hw.physicalcpu)
 fi
 
 # Discover clang-format
 if type clang-format-10 2> /dev/null ; then
     CLANG_FORMAT=clang-format-10
-elif type clang-format-8 2> /dev/null ; then
-    CLANG_FORMAT=clang-format-8
-else
+elif type clang-format 2> /dev/null ; then
+    # Clang format found, but need to check version
     CLANG_FORMAT=clang-format
+    V=$(clang-format --version)
+    if [[ $V != "*version 10.0*" ]]; then
+        echo "clang-format is not 10.0 (returned ${V})"
+        exit 1
+    fi
+else
+    echo "No appropriate clang-format found (expected clang-format-10.0.0, or clang-format)"
+    exit 1
 fi
 
-find . -type d \( -path ./deps \
--o -path ./cmake \
--o -path ./plugins/decklink/win/decklink-sdk \
--o -path ./plugins/decklink/mac/decklink-sdk \
--o -path ./plugins/decklink/linux/decklink-sdk \
--o -path ./plugins/enc-amf \
--o -path ./plugins/mac-syphon/syphon-framework \
--o -path ./plugins/obs-outputs/ftl-sdk \
--o -path ./plugins/obs-vst \
--o -path ./build \) -prune -type f -o -name '*.h' -or -name '*.hpp' -or -name '*.m' -or -name '*.mm' -or -name '*.c' -or -name '*.cpp' \
-| xargs -L100 -P${NPROC} ${CLANG_FORMAT} -i -style=file  -fallback-style=none
+find . -type d \( \
+    -path ./\*build -o \
+    -path ./cmake -o \
+    -path ./deps -o \
+    -path ./plugins/decklink/\*/decklink-sdk -o \
+    -path ./plugins/enc-amf -o \
+    -path ./plugins/mac-syphon/syphon-framework -o \
+    -path ./plugins/obs-outputs/ftl-sdk -o \
+    -path ./plugins/obs-vst \
+\) -prune -false -type f -o \
+    -name '*.h' -or \
+    -name '*.hpp' -or \
+    -name '*.m' -or \
+    -name '*.m,' -or \
+    -name '*.c' -or \
+    -name '*.cpp' \
+ | xargs -L100 -I{} -P ${NPROC} ${CLANG_FORMAT} ${VERBOSITY} -i -style=file -fallback-style=none {}
```

Comment:
I get this warning when using the new `formatcode.sh`
```
xargs: warning: options --max-lines and --replace/-I/-i are mutually exclusive, ignoring previous --max-lines value
```

Reply:
Which clang-format version? Had no issues with v10.

Reply:
xargs (GNU findutils) 4.8.0

Reply:
Removed the `-I`.